### PR TITLE
Secure Data Path support in pseudo-TAs

### PR DESCRIPTION
**Pull request by etienne-lms**
Restore SDP support in pseudo Trusted Applications.

This PR is extracted from https://github.com/OP-TEE/optee_os/pull/1322 and supersedes it.

### CODE REVIEW

**Comment by etienne-lms** [comment]
This PR is NOT applicable as is. It should be rebased on top of https://github.com/OP-TEE/optee_os/pull/1412. Yet, i created it straight to open up reviews.

File: core/arch/arm/pta/pta_invoke_test.c [hunk 1]
```diff
@@ -36,9 +36,9 @@
 
 #include "core_self_tests.h"
 
-#define TA_NAME		"sta_self_tests.ta"
+#define TA_NAME		"invoke_test.pta"
```
Comment (jforissier) [inline core/arch/arm/pta/pta_invoke_test.c]: Minor: why not keep the pural? (`invoke_tests.pta`) for consistency with other tests, see `sub.mk`.

Comment (etienne-lms) [inline core/arch/arm/pta/pta_invoke_test.c]: agree, i'll change the string and the file name.

**Comment by etienne-lms** [comment]
i will rebase this change since the cache support is now merge and i needed it for SDP pTAs.
So rebase + an extra commit for SDP test pTA (the invoke_test**s** pTA) to manipulate cache.

**Comment by etienne-lms** [comment]
rebase since this PR had dependencies on https://github.com/OP-TEE/optee_os/pull/1412 that is now merged.
Travis should no more complain...

File: lib/libutee/include/pta_invoke_tests.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+/*
+ * Types of parameter drives the test sequences:
+ * - test on value parameters
+ * - test on SHM memory reference parameters
+ * - test on SDP memory reference parameters
+ */
+#define PTA_INVOKE_TESTS_CMD_PARAMS		1
+
+/* don't care about parameters, run some core internal tests. */
```
Comment (jforissier) [inline lib/libutee/include/pta_invoke_tests.h]: `/* Run some core internal tests */`

File: lib/libutee/include/pta_invoke_tests.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#define PTA_INVOKE_TESTS_CMD_SELF_TESTS		2
+
+/*
+ * [in]		memref[0]	input data to be read
+ * [out]	memref[1]	output buffer to be filled
+ */
+#define PTA_INVOKE_TESTS_CMD_INJECT_SDP		3
```
Comment (jforissier) [inline lib/libutee/include/pta_invoke_tests.h]:
```C
/*
 * Secure data path: check that pTA can copy data from non-secure shared memory
 * to SDP secure memory
 *
 * [in]     memref[0]        source (non-secure shared memory)
 * [out]    memref[1]        destination (SDP secure memory)
 */
#define PTA_INVOKE_TEST_SDP_COPY_NSEC_TO_SEC 3
```

File: lib/libutee/include/pta_invoke_tests.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+/*
+ * [in]		memref[0]	input data to be read
+ * [out]	memref[0]	output buffer to be filled
+ */
+#define PTA_INVOKE_TESTS_CMD_TRANSFORM_SDP	4
```
Comment (jforissier) [inline lib/libutee/include/pta_invoke_tests.h]:
```C
/*
 * Secure data path: check that pTA can read data from SDP secure memory and
 * write it back
 *
 * [in/out]     memref[0]        SDP secure buffer to read from and write to
 */
#define PTA_INVOKE_TEST_SDP_READ_WRITE        4
```

File: lib/libutee/include/pta_invoke_tests.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+/*
+ * [in]		memref[0]	input data to be read
+ * [out]	memref[1]	output buffer to be filled
+ */
+#define PTA_INVOKE_TESTS_CMD_DUMP_SDP		5
```
Comment (jforissier) [inline lib/libutee/include/pta_invoke_tests.h]:
```C
/*
 * Secure data path: check that pTA can copy data from SDP secure memory to
 * non-secure shared memory
 *
 * [in]     memref[0]        source (SDP secure memory)
 * [out]    memref[1]        destination (non-secure shared memory)
 */
#define PTA_INVOKE_TEST_SDP_COPY_SEC_TO_NSEC 5
```

**Comment by jenswi-linaro** [comment]
With Jeromes comments addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: lib/libutee/include/pta_invoke_tests.h (hunk 2 above)
Comment (etienne-lms) [inline lib/libutee/include/pta_invoke_tests.h]: ok

File: lib/libutee/include/pta_invoke_tests.h (hunk 3 above)
Comment (etienne-lms) [inline lib/libutee/include/pta_invoke_tests.h]: ok

File: lib/libutee/include/pta_invoke_tests.h (hunk 4 above)
Comment (etienne-lms) [inline lib/libutee/include/pta_invoke_tests.h]: ok

**Comment by etienne-lms** [comment]
comments addressed. i will rebase, squash and apply tags...

**Comment by jforissier** [comment]
Thanks @etienne-lms!

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
rebased, squashed, tags applied.

**Pull request merged**